bundles/org.openhab.binding.tibber/src/main/java/org/openhab/binding/tibber/internal/handler/TibberHandler.java

```diff
@@ -0,0 +1,471 @@
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingStatusInfo;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.openhab.binding.tibber.internal.config.TibberConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link TibberHandler} is responsible for handling queries to/from Tibber API.
+ *
+ * @author Stian Kjoglum - Initial contribution
+ */
+@NonNullByDefault
+public class TibberHandler extends BaseThingHandler {
+    private static final int REQUEST_TIMEOUT = (int) TimeUnit.SECONDS.toMillis(20);
+    private Logger logger = LoggerFactory.getLogger(TibberHandler.class);
+    private final Properties httpHeader = new Properties();
+    private SslContextFactory sslContextFactory = new SslContextFactory(true);
+    private @Nullable TibberWebSocketListener socket;
+    private @Nullable Session session;
+    private boolean connected = false;
+    private String rtEnabled = "false";
+
+    private @Nullable ScheduledFuture<?> pollingJob;
+
+    public TibberHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+
+        getTibberParameters();
+        TibberConfiguration tibberConfig = getConfigAs(TibberConfiguration.class);
+        startRefresh(tibberConfig.getRefresh());
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+
+        if (command instanceof RefreshType) {
+            TibberConfiguration tibberConfig = getConfigAs(TibberConfiguration.class);
+            startRefresh(tibberConfig.getRefresh());
+
+        } else {
+            logger.debug("Tibber API is read-only and does not handle commands");
+        }
+    }
+
+    public void getTibberParameters() {
+        try {
+            httpHeader.put("cache-control", "no-cache");
+            httpHeader.put("content-type", JSON_CONTENT_TYPE);
+
+            TibberConfiguration tibberConfig = getConfigAs(TibberConfiguration.class);
+            String token = new StringBuilder("Bearer ").append(tibberConfig.getToken()).toString();
+            httpHeader.put("Authorization", token);
+
+            TibberPriceConsumptionHandler tibberQuery = new TibberPriceConsumptionHandler();
+            InputStream connectionStream = tibberQuery.connectionInputStream(tibberConfig.getHomeid());
+            String response = HttpUtil.executeUrl("POST", BASE_URL, httpHeader, connectionStream, null,
+                    REQUEST_TIMEOUT);
+
+            if (!response.contains("error")) {
+                logger.info("Initialized ID: {}", tibberConfig.getHomeid());
```

Comment:
Remove this or lower the logging level. The user is already notified from you switching the status to online anyway.

Reply:
Removed.